sa/database_test.go

```diff
@@ -73,3 +73,23 @@ func TestStrictness(t *testing.T) {
 		t.Fatalf("Got wrong type of error: %s", err)
 	}
 }
+
+func TestTimeouts(t *testing.T) {
+	dbMap, err := NewDbMap(vars.DBConnSA+"?readTimeout=100ms", 1)
+	if err != nil {
+		t.Fatal("Error setting up DB:", err)
+	}
+	// SLEEP is defined to return 1 if it was interrupted, but we want to actually
+	// get an error to simulate what would happen with a slow query. So we wrap
+	// the SLEEP in a subselect.
+	_, err = dbMap.Exec(`SELECT 1 FROM (SELECT SLEEP(1)) as subselect;`)
+	if err == nil {
+		t.Fatal("Expected error when running slow query, got none.")
+	}
+
+	// We expect to get:
+	// Error 1969: Query execution was interrupted (max_statement_time exceeded)
+	if !strings.Contains(err.Error(), "Error 1967") {
```

Comment:
I think this should be `Error 1969` based on the [error code docs](https://mariadb.com/kb/en/mariadb/mariadb-error-codes/).